tools/cp-caps/rcp_caps_test.py

```diff
@@ -79,9 +79,44 @@ def test_diag_commands(self):
         self.__ref.diag_stop()
         self.__dut.diag_stop()
 
+    def test_csl(self):
+        self.__dataset = self.__get_default_dataset()
+        self.__test_csl_transmitter()
+
     #
     # Private methods
     #
+    def __get_default_dataset(self):
+        return self.__dut.create_dataset(channel=20, network_key='00112233445566778899aabbccddcafe')
+
+    def __test_csl_transmitter(self):
+        packets = 10
+
+        self.__dut.factory_reset()
+        self.__ref.factory_reset()
+
+        self.__dut.join(self.__dataset)
+        self.__dut.wait_for('state', 'leader')
+
+        # Set the reference device as an SSED
+        self.__ref.set_mode('-')
+        self.__ref.config_csl(channel=15, period=320000, timeout=100)
+        self.__ref.join(self.__dataset)
+        self.__ref.wait_for('state', 'child')
+
+        child_table = self.__dut.get_child_table()
+        ret = len(child_table) == 1 and child_table[1]['csl']
+
+        if ret:
+            ref_mleid = self.__ref.get_ipaddr_mleid()
+            result = self.__dut.ping(ref_mleid, count=packets, interval=1)
+            ret = result['transmitted_packets'] == result['received_packets'] == packets
```

Comment:
Nice! I did not know we can do multiple `==` to check if all variables are same in python. 😃